File: angularapp/src/app/services/property.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Property } from '../models/property.model';

@Injectable({
  providedIn: 'root'
})
export class PropertyService {
  private readonly backendUrl="http://localhost:8080/api/properties"
  constructor(private readonly http:HttpClient) { }

  getAllProperties():Observable<Property[]>{
    return this.http.get<Property[]>(this.backendUrl);
  }

  getPropertyById(propertyId:number):Observable<Property>{
    return this.http.get<Property>(`${this.backendUrl}/${propertyId}`);
  }

  addProperty(property:Property):Observable<Property>{
    return this.http.post<Property>(this.backendUrl,property);
  }

  updateProperty(propertyId:number,property:Property):Observable<Property>{
    return this.http.put<Property>(`${this.backendUrl}/${propertyId}`,property);
  }

  deleteProperty(propertyId:number):Observable<void>{
    return this.http.delete<void>(`${this.backendUrl}/${propertyId}`);
  }
  
}

File: springapp/src/main/java/com/examly/springapp/service/PropertyService.java
package com.examly.springapp.service;

import java.util.List;
import java.util.Optional;

import com.examly.springapp.model.Property;

public interface PropertyService {
    Property addProperty(Property property);
    Optional<Property> getPropertyById(long propertyId);
    List<Property> getAllProperties();

    Property updateProperty(long propertyId,Property property);
    void deleteProperty(long propertyId);

}
